library/core/src/pin.rs

```diff
@@ -1003,22 +1003,26 @@ impl<P, U> CoerceUnsized<Pin<U>> for Pin<P> where P: CoerceUnsized<U> {}
 #[stable(feature = "pin", since = "1.33.0")]
 impl<P, U> DispatchFromDyn<Pin<U>> for Pin<P> where P: DispatchFromDyn<U> {}
 
-/// Constructs a <code>[Pin]<[&mut] T></code>, by pinning[^1] a `value: T` _locally_[^2].
+/// Constructs a <code>[Pin]<[&mut] T></code>, by pinning a value locally.
 ///
-/// Unlike [`Box::pin`], this does not involve a heap allocation.
+/// Unlike [`Box::pin`], this does not involve a heap allocation. As explained
+/// below, the element might still end up on the heap however.
```

Comment:
This builds along the lines of my "don't talk about heap/stack" suggestion elsewhere. Mentioning the allocation may be on the heap is a distraction if what we _really_ want to say is that it's not creating _another_ allocation.
```suggestion
/// Unlike [`Box::pin`], this does not involve a new allocation.
```